### PR TITLE
[SPARK-12536] [SQL] Added "[]" in Explain Outputs For Empty LocalRelation and LocalTableScan

### PR DESCRIPTION
**Pull request by gatorsmile**
The filter `Filter (False)` generates an empty `LocalRelation` in `SimplifyFilters`. In the current `Explain`, the optimized and physical plans look confusing. 

For example,

``` scala
val df = Seq(1 -> "a").toDF("a", "b")
df.where("1 = 0").explain(true)
```

```
== Analyzed Logical Plan ==
a: int, b: string
Filter (1 = 0)
+- Project [_1#0 AS a#2,_2#1 AS b#3]
   +- LocalRelation [_1#0,_2#1], [[1,a]]

== Optimized Logical Plan ==
LocalRelation [a#2,b#3] 

== Physical Plan ==
LocalTableScan [a#2,b#3] 
```

After the fix, the Optimized Logical Plan and Physical Plan are changed to

```
== Optimized Logical Plan ==
LocalRelation [a#2,b#3] []

== Physical Plan ==
LocalTableScan [a#2,b#3] [] 
```

The original PR is: https://github.com/apache/spark/pull/10494

### CODE REVIEW

**Comment by gatorsmile** [comment]
@cloud-fan @marmbrus Hopefully, my understanding is right. Feel free to let me know if the fix needs any change. Thank you!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala [hunk 1]
```diff
@@ -398,6 +395,7 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
   def argString: String = productIterator.flatMap {
     case tn: TreeNode[_] if containsChild(tn) => Nil
     case tn: TreeNode[_] => s"${tn.simpleString}" :: Nil
+    case seq: Seq[BaseType] if seq.isEmpty => "[]" :: Nil
     case seq: Seq[BaseType] if seq.toSet.subsetOf(children.toSet) => Nil
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala]: I think a better way is to change this `if` to `if seq.nonEmpty && seq.toSet.subsetOf(children.toSet)`

Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala]: Thank you! Will do!

**Comment by gatorsmile** [comment]
retest this please.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala [hunk 2]
```diff
@@ -26,17 +26,14 @@ import org.json4s.JsonAST._
...
+import org.apache.spark.SparkContext
 import org.apache.spark.sql.catalyst.ScalaReflection._
+import org.apache.spark.sql.catalyst.ScalaReflectionLock
 import org.apache.spark.sql.catalyst.errors._
 import org.apache.spark.sql.catalyst.expressions._
-import org.apache.spark.sql.catalyst.plans.logical.Statistics
-import org.apache.spark.sql.catalyst.util.DateTimeUtils
 import org.apache.spark.sql.types._
 import org.apache.spark.storage.StorageLevel
-import org.apache.spark.unsafe.types.{CalendarInterval, UTF8String}
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala]: these import changes are irrelevant？

Comment (gatorsmile) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala]: Yeah, I just saw these useless imports in this file. Do you want me to keep them unchanged?

Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala]: oh, it's fine to remove useless imports.

**Comment by cloud-fan** [comment]
LGTM

**Comment by gatorsmile** [comment]
Thank you for your help! Sorry for wasting you a lot of time on it! @cloud-fan

**Comment by rxin** [comment]
@gatorsmile I'm actually not sure whether the new format is more clear. Do you mind closing this? Thanks.

**Comment by gatorsmile** [comment]
Sure, let me close it.

**Pull request closed without merging**